### PR TITLE
Rough Background Reconcile

### PR DESCRIPTION
**Pull request by praymann**
Let me first say there is a lot I don't like here...

But!

It adds:
* Two ActiveJob(s), one to reconcile a given Software which is fired upon creation and one to reconcile all Software in batches
* Adds uniqueness validation for name/org pairs

It doesn't:
* Move into multi-Models 🤷‍♂️ 
* Have actual tests
* Wire in an actual queue for ActiveJob for "production", it's in-memory on localhost

So it's a start that I'll improve.

### CODE REVIEW

File: db/migrate/20200411224202_add_in_use_release_to_software.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddInUseReleaseToSoftware < ActiveRecord::Migration[5.2]
+  def change
+    add_column :softwares, :in_use_release, :string
+    add_column :softwares, :in_use_release_date, :date
```
Comment (jadametz) [inline db/migrate/20200411224202_add_in_use_release_to_software.rb:4]: How come the fields from the two proposed migrations aren't represented in the model?

Comment (praymann) [reply db/migrate/20200411224202_add_in_use_release_to_software.rb:4]: `ActiveRecord::Base` doesn't require explicit attribute definitions, it always infers them from the table. When you want validators/call_backs/etc is when you add them to the model explicitly.

**Comment by jadametz** [comment]
> So it's a start that I'll improve.

@praymann any plans to work on this PR more before I merge it to utilize its functionality?

**Comment by jadametz** [comment]
@praymann I'm going to merge this so that we can bring in `rspec` and the schema / UI changes.

**Pull request merged**